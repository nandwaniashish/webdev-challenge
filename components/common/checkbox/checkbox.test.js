import React from 'react'
import { render, mount } from 'enzyme'
import Checkbox from './index'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() }) //TODO: move this to jest config
describe('checkbox component', () => {
  const sampleLabel = 'Sample label'
  it('renders default', () => {
    const wrapper = render(
      <Checkbox id="check-test-1" label={sampleLabel}></Checkbox>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('calls onChangeHandler if checked', () => {
    const onMockChange = jest.fn()
    const wrapper = mount(
      <Checkbox
        id="hello"
        label="whi"
        checked={false}
        onChange={onMockChange}
      ></Checkbox>
    )
    wrapper.find('input').simulate('change')

    expect(onMockChange).toHaveBeenCalled()
  })
})
