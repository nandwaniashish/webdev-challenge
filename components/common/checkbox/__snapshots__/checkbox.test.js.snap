// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox component renders default 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "id": "check-test-1",
          "name": "check-test-1",
          "type": "checkbox",
          "value": "",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "checkboxLabel",
            "for": "check-test-1",
          },
          "children": Array [
            Node {
              "data": "Sample label",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "label",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "for": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "for": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "id": undefined,
          "name": undefined,
          "type": undefined,
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "id": undefined,
          "name": undefined,
          "type": undefined,
          "value": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "checkboxLabel",
          "for": "check-test-1",
        },
        "children": Array [
          Node {
            "data": "Sample label",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "label",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "id": "check-test-1",
            "name": "check-test-1",
            "type": "checkbox",
            "value": "",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "id": undefined,
            "name": undefined,
            "type": undefined,
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "id": undefined,
            "name": undefined,
            "type": undefined,
            "value": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "for": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "for": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
